cpp/turbodbc_arrow/Test/tests/arrow_result_set_test.cpp

```diff
@@ -0,0 +1,436 @@
+	cpp_odbc::multi_value_buffer buffer(sizeof(int64_t), OUTPUT_SIZE);
+  memcpy(buffer.data_pointer(), typed_array->data()->data(), sizeof(int64_t) * OUTPUT_SIZE);
+	std::vector<std::reference_wrapper<cpp_odbc::multi_value_buffer const>> expected_buffers = {buffer};
+	EXPECT_CALL(rs, do_get_buffers()).WillOnce(testing::Return(expected_buffers));
+  EXPECT_CALL(rs, do_fetch_next_batch()).WillOnce(testing::Return(OUTPUT_SIZE)).WillOnce(testing::Return(0));
+
+  turbodbc_arrow::arrow_result_set ars(rs);
+  std::shared_ptr<arrow::Table> table;
+  ASSERT_OK(ars.fetch_all_native(&table));
+  ASSERT_TRUE(expected_table->Equals(*table));
+}
+
+TEST(ArrowResultSetTest, MultipleBatchSingleColumnResultSetConversion)
+{
+	mock_result_set rs;
+  const int64_t OUTPUT_SIZE = 100;
+
+  // Expected output: a Table with a single column of int64s
+  arrow::Int64Builder builder(arrow::default_memory_pool(), arrow::int64());
+  for (int64_t i = 0; i < 2 * OUTPUT_SIZE; i++) {
+    ASSERT_OK(builder.Append(i));
+  }
+  std::shared_ptr<arrow::Array> array;
+  ASSERT_OK(builder.Finish(&array));
+  std::shared_ptr<arrow::Int64Array> typed_array = std::static_pointer_cast<arrow::Int64Array>(array);
+  std::vector<std::shared_ptr<arrow::Field>> fields({std::make_shared<arrow::Field>("int_column", arrow::int64(), true)});
+  std::shared_ptr<arrow::Schema> schema = std::make_shared<arrow::Schema>(fields);
+  std::vector<std::shared_ptr<arrow::Column>> columns ({std::make_shared<arrow::Column>(fields[0], array)});
+  std::shared_ptr<arrow::Table> expected_table = std::make_shared<arrow::Table>(schema, columns);
+
+  // Mock schema
+	std::vector<turbodbc::column_info> expected = {{
+    "int_column", turbodbc::type_code::integer, true}};
+	EXPECT_CALL(rs, do_get_column_info()).WillRepeatedly(testing::Return(expected));
+
+  // Mock output columns
+  // * Two batches of 100 ints
+	cpp_odbc::multi_value_buffer buffer_1(sizeof(int64_t), OUTPUT_SIZE);
+  memcpy(buffer_1.data_pointer(), typed_array->data()->data(), sizeof(int64_t) * OUTPUT_SIZE);
+	std::vector<std::reference_wrapper<cpp_odbc::multi_value_buffer const>> expected_buffers_1 = {buffer_1};
+	cpp_odbc::multi_value_buffer buffer_2(sizeof(int64_t), OUTPUT_SIZE);
+  memcpy(buffer_2.data_pointer(), typed_array->data()->data() + sizeof(int64_t) * OUTPUT_SIZE, sizeof(int64_t) * OUTPUT_SIZE);
+	std::vector<std::reference_wrapper<cpp_odbc::multi_value_buffer const>> expected_buffers_2 = {buffer_2};
+	EXPECT_CALL(rs, do_get_buffers()).WillOnce(testing::Return(expected_buffers_1)).WillOnce(testing::Return(expected_buffers_2));
+  EXPECT_CALL(rs, do_fetch_next_batch()).WillOnce(testing::Return(OUTPUT_SIZE)).WillOnce(testing::Return(OUTPUT_SIZE)).WillOnce(testing::Return(0));
+
+  turbodbc_arrow::arrow_result_set ars(rs);
+  std::shared_ptr<arrow::Table> table;
+  ASSERT_OK(ars.fetch_all_native(&table));
+  ASSERT_TRUE(expected_table->Equals(*table));
+}
+
+TEST(ArrowResultSetTest, MultipleBatchMultipleColumnResultSetConversion)
```

Comment:
The other tests are pretty long already. This one approaches 300 lines! Please, split it up into sensible subtests with sensible names. Use helper functions to extract common setup logic where it makes sense.